packages/error-reporting/src/google-apis/auth-client.js

```diff
@@ -88,26 +89,46 @@ class RequestHandler extends common.Service {
    */
   constructor(config, logger) {
     var pid = config.getProjectId();
+    // If an API key is provided, do not try to authenticate.
+    var tryAuthenticate = !config.getKey();
     super({
       packageJson: pkg,
-      baseUrl: 'https://clouderrorreporting.googleapis.com/v1beta1/',
+      baseUrl: API,
       scopes: SCOPES,
       projectId: pid !== null ? pid : undefined,
-      projectIdRequired: true
+      projectIdRequired: true,
+      customEndpoint: !tryAuthenticate
     }, config);
     this._config = config;
     this._logger = logger;
 
     var that = this;
-    this.authClient.getToken(function(err, token) {
-      if (err) {
-        that._logger.error([
-          'Unable to find credential information on instance. This library',
-          'will be unable to communicate with the Stackdriver API to save',
-          'errors.  Message: ' + err.message
-        ].join(' '));
-      }
-    });
+    if (tryAuthenticate) {
+      this.authClient.getToken(function(err, token) {
+        if (err) {
+          that._logger.error([
+            'Unable to find credential information on instance. This library',
+            'will be unable to communicate with the Stackdriver API to save',
+            'errors.  Message: ' + err.message
+          ].join(' '));
+        }
+      });
+    } else {
+      this.request({
+        uri: 'events:report',
+        qs: RequestHandler.manufactureQueryString(this._config.getKey()),
+        method: 'POST',
+        json: {}
+      }, (err, body, response) => {
```

Comment:
@kjin Should we also check `response.statusCode` to be 400?

@0x4b Does this seem like an appropriate way to check if the API key is valid? Is there a test on the service side that ensures that 400 + 'Message cannot be empty.' is not returned when the key is invalid?

Reply:
There isn't such a test, largely because some auth related error cases are handled by a different part of the stack outside the service's control.